core/trino-spi/src/main/java/io/trino/spi/expression/StandardFunctions.java

```diff
@@ -76,5 +76,10 @@ private StandardFunctions() {}
      */
     public static final FunctionName NEGATE_FUNCTION_NAME = new FunctionName("$negate");
 
+    /**
+     * $between is a function accepting three arguments - value, min and max
+     */
+    public static final FunctionName BETWEEN_FUNCTION_NAME = new FunctionName("$between");
```

Comment:
Do we need a new function?
I would prefer to model this as `$and(x >= y, x <= z)`.

BETWEEN has important property that it evaluates the LHS value once, but that doesn't matter for deterministic predicates, and we don't want to handle non-deterministic predicates in connector expressions.

(That's also why we could model boolean AND as an ordinary function. Lazy-evaluation (or lack thereof) isn't defined for connector expressions.)

Reply:
After writing this PR, I've noticed that `BETWEEN` is already modeled the way you said at `DomainTranslator`, so the implementation will only be reflected in the projection pushdown. Do you think I should implement the conversion to an `$and` right now or just wait with this PR?

Reply:
What for would you want to wait for here?

Reply:
I thought I might need to wait for a connector that will actually use this, but seems like we're ok with just adding this (with tests) so I'll continue

Reply:
PostgreSQL would use this already, so you can try add tests there.
see `io.trino.plugin.postgresql.TestPostgreSqlConnectorTest#test*PredicatePushdown`

Reply:
`GREATER_THAN_OR_EQUAL` and `LESS_THAN_OR_EQUAL` are currently not supported in PostgreSQL: https://github.com/trinodb/trino/blob/8b6888c33707ff72636882f671766f48a6e6eaa9/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java#L302-L303

Reply:
i guess we should just add them there?

Reply:
cc @hashhar on the above

in any case, let's split the engine part (handling of BETWEEN)
and PostgreSQL part (handling of inequalities) into separate PRs.

thanks